crates/external_services/src/grpc_client/health_check_client.rs

```diff
@@ -0,0 +1,149 @@
+use std::{collections::HashMap, fmt::Debug};
+
+use api_models::health_check::{HealthCheckMap, HealthCheckServices};
+use common_utils::{errors::CustomResult, ext_traits::AsyncExt};
+use error_stack::ResultExt;
+pub use health_check::{
+    health_check_response::ServingStatus, health_client::HealthClient, HealthCheckRequest,
+    HealthCheckResponse,
+};
+use router_env::logger;
+
+#[allow(
+    missing_docs,
+    unused_qualifications,
+    clippy::unwrap_used,
+    clippy::as_conversions,
+    clippy::use_self
+)]
+pub mod health_check {
+    tonic::include_proto!("grpc.health.v1");
+}
+
+use super::{Client, DynamicRoutingClientConfig, GrpcClientSettings};
+
+/// Result type for Dynamic Routing
+pub type HealthCheckResult<T> = CustomResult<T, HealthCheckError>;
+/// Dynamic Routing Errors
+#[derive(Debug, Clone, thiserror::Error)]
+pub enum HealthCheckError {
+    /// The required input is missing
+    #[error("Missing fields: {0} for building the Health check connection")]
+    MissingFields(String),
+    /// Error from gRPC Server
+    #[error("Error from gRPC Server : {0}")]
+    ConnectionError(String),
+    /// status is invalid
+    #[error("Invalid Status from server")]
+    InvalidStatus,
+}
+
+/// Health Check Client type
+#[derive(Debug, Clone)]
+pub struct HealthCheckClient {
+    /// Health clients for all gRPC based services
+    pub clients: HashMap<HealthCheckServices, HealthClient<Client>>,
```

Comment:
`HealthClient` is specific to dynamic routing health check client right? other grpc services's health client cannot be added to this HashMap as its HealthClient will be different

Reply:
If I understand this correctly, the `HealthClient` type is borrowed from proto types, in which case the type for the `HealthClient` for all other gRPC services should be the same (protofile being the same). The `HealthClient` instance however, would be different for each service in the hashmap.